src/esp/sim/Simulator.cpp

```diff
@@ -809,5 +821,168 @@ void Simulator::setObjectLightSetup(int objectID,
   }
 }
 
+int Simulator::findNearestObject(int refObjectID, float distance) {
+  float minDist = 1.0;
+  int nearestObjectID = -1; 
+  if (refObjectID >= 0) {
+    Magnum::Vector3 refPosition = 
+        physicsManager_->getTranslation(refObjectID);
+    Magnum::Vector2 agentPos_xz(refPosition.x(), refPosition.z());
+
+    for(int objectID : physicsManager_->getExistingObjectIDs()) {
+      if (objectID != refObjectID) {
+        Magnum::Vector3 objectPos = physicsManager_->getTranslation(objectID);
+        Magnum::Vector2 objectPosXZ(objectPos.x(), objectPos.z());
+        
+        float d = Magnum::Vector2(agentPos_xz - objectPosXZ).length();
```

Comment:
Wait, so the distance does not include y? This is bad considering we have datasets with multi-floor environments.

Reply:
If we assume object COMs are within 0.5m in Y of the NavMesh, geodesic distance could be better 🤷‍♂️ . May be some caveats to this approach also.

Reply:
@Skylion007  one of the `findNearestObject` is not being used. It was only used for testing purpose. I'll remove the code for that.